src/components/safe-messages/MsgModal/index.tsx

```diff
@@ -80,16 +86,13 @@ const MsgModal = ({
     return messageHash ?? generateSafeMessageHash(safe, decodedMessage)
   }, [messageHash, safe, decodedMessage])
 
-  // Get already proposed message
-  const [alreadyProposedMessage] = useAsync<SafeMessage | Omit<SafeMessage, 'type'>>(() => {
-    const localMessage = messages.page?.results
-      .filter(isSafeMessageListItem)
-      .find((message) => message.messageHash === messageHash)
+  const existingMessage = messages.page?.results
```

Comment:
If we want to use these messages instead of polling I would suggest reducing the polling interval.
Currently we only poll every 15 seconds. That's a possibly long time to wait. Especially if it would be a 1/n Safe.